storage/flush.go

```diff
@@ -100,16 +140,51 @@ func (m *flushManager) Flush(curr time.Time) error {
 
 func (m *flushManager) Report() {
 	m.RLock()
-	flushInProgress := m.flushInProgress
+	state := m.state
 	m.RUnlock()
 
-	if flushInProgress {
-		m.status.Update(1)
+	if state == flushManagerFlushInProgress {
+		m.isFlushing.Update(1)
+	} else {
+		m.isFlushing.Update(0)
+	}
+
+	if state == flushManagerSnapshotInProgress {
+		m.isSnapshotting.Update(1)
 	} else {
-		m.status.Update(0)
+		m.isSnapshotting.Update(0)
 	}
 }
 
+func (m *flushManager) setFlushInProgress(b bool) {
+	m.Lock()
+	m.state = flushManagerFlushInProgress
+	m.Unlock()
+}
+func (m *flushManager) setSnapshotInProgress(b bool) {
```

Comment:
Hm this is strange, the boolean isn't used anymore?  Can we remove it and only call this from the places where we were passing `true` to this call?

Reply:
refactored into a setState method thats shared